src/types/frozenset.rs

```diff
@@ -271,6 +372,25 @@ mod tests {
         });
     }
 
+    #[test]
+    fn test_frozenset_iter_size_hint() {
+        Python::with_gil(|py| {
+            let set = PyFrozenSet::new(py, &[1]).unwrap();
+
+            let mut iter = set.iter();
+
+            if cfg!(Py_LIMITED_API) {
+                assert_eq!(iter.size_hint(), (0, None));
+            } else {
+                assert_eq!(iter.len(), 1);
```

Comment:
This fails to compile as both arms of the if clause are still built even if only one is actually reachable, i.e. this needs to use blocks or maybe even just separate test case definitions.